internal/parse/parser.go

```diff
@@ -118,15 +119,15 @@ func (p *Parser) add(part queryPart) {
 }
 
 // Parse takes an input string and parses the input and output parts. It returns
-// a pointer to a ParsedExpr.
+// a pointer to a ParsedExpr. If the parser encounters an error then ParsedExpr
+// is nil.
 func (p *Parser) Parse(input string) (expr *ParsedExpr, err error) {
 	defer func() {
 		if err != nil {
 			err = fmt.Errorf("cannot parse expression: %s", err)
 		}
 	}()
 	p.init(input)
-
```

Comment:
A little breathing room is good, perhaps a blank line above instead would look nice?